src/devices/computrainerbike/computrainerbike.cpp

```diff
@@ -183,10 +161,8 @@ void computrainerbike::innerWriteResistance() {
     bool erg_mode = settings.value(QZSettings::zwift_erg, QZSettings::default_zwift_erg).toBool();
 
     if (requestResistance != -1) {
-        if (requestResistance > max_resistance) {
-            requestResistance = max_resistance;
-        } else if (requestResistance < min_resistance) {
-            requestResistance = min_resistance;
+        if(!this->resistanceLimits().contains(requestResistance)) {
+            requestResistance = resistanceLimits().clip(requestResistance);
         } else if (requestResistance == 0) {
             requestResistance = 1;
         }
```

Comment:
@cagnulein What's going on with these few bikes that have this form of clipping of the resistance request?

 i.e. here the min_resistance constant is -20 and the max_resistance constant is 100, and you clip the requestedResistance to that range, except if it's 0 then you make it 1.

Reply:
This is also in the proformwifi and proformtelnet bikes.

Reply:
min_resistance can't never be less than 0 in QZ. I mean I never encountered a bike with a negative resistance. So the current master can't handle this. That's why for inclination, instead, i'm using a -100 value as a "no action" value (because inclination can go lower than 0 (but not to -100 of course)).

did I answer to your question?

Reply:
I'm aware of the use of -1 and -100 to specify "no action" for resistance and inclination. But I don't know the purpose of the -20..100 range you're using here. I'll use 1..100 for this exercise unless you specify something else.

BTW, you might have noticed in some test code I've switched to C++17 and used the std::optional<T> template. This enables you to specify something like
```
std::optional<resistance_t> resistanceRequest;
```
and use resistanceRequest.hasValue() and resistanceRequest.reset() instead of have a "no action constant". If you are agreeable, I will do this for the resistance, inclination and power requests, in a different PR.

Reply:
(moved to a separate message so you can approve the whole message with a thumbs up)

BTW, you might have noticed in some test code I've switched to C++17 and used the std::optional<T> template. This enables you to specify something like
```
std::optional<resistance_t> resistanceRequest;
```
and use resistanceRequest.has_value() and resistanceRequest.reset() instead of having a "no action constant". If you are agreeable, I will do this for the resistance, inclination and power requests, in a different PR.

Reply:
sorry for the delay @drmason789 

the -20 in this case was copied by another project that I found so actually I didn't care about it. It could be that this bike can have also negative resistance (so it will make sense to use -100 for the resistance too). I didn't remember this.

> and use resistanceRequest.has_value() and resistanceRequest.reset() instead of having a "no action constant". If you are agreeable, I will do this for the resistance, inclination and power requests, in a different PR.

yes sure! thanks!